src/frontend/xgboost_json.cc

```diff
@@ -411,7 +411,9 @@ bool LearnerHandler::StartObject() {
           push_key_handler<GradientBoosterHandler, treelite::ModelImpl<float, float>>(
               "gradient_booster", *output.model) ||
           push_key_handler<ObjectiveHandler, std::string>("objective", objective) ||
-          push_key_handler<IgnoreHandler>("attributes"));
+          push_key_handler<IgnoreHandler>("attributes") ||
+          push_key_handler<IgnoreHandler>("feature_names") ||
+          push_key_handler<IgnoreHandler>("feature_types"));
```

Comment:
Somehow I am not able to register more than one `IgnoreHandler` with different keys. Will troubleshoot.

Reply:
Error trace:
```
E           treelite.util.TreeliteError: [21:27:24] ../src/frontend/xgboost_json.cc:515: Provided JSON could not be parsed as XGBoost model. Parsing error at offset 45: Terminate parsing due to Handler error.
E           {"learner":{"attributes":{},"feature_names":[],"feature_types":[],"gradient_booster":{"model":{"gbtr
E           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
E           Stack trace:
E             [bt] (0) /home/phcho/Desktop/treelite/python/treelite/../../build/libtreelite.so(dmlc::LogMessageFatal::~LogMessageFatal()+0x82) [0x7f0a4d702422]
E             [bt] (1) /home/phcho/Desktop/treelite/python/treelite/../../build/libtreelite.so(treelite::frontend::LoadXGBoostJSONModel(char const*)+0x86e) [0x7f0a4d7abc5e]
E             [bt] (2) /home/phcho/Desktop/treelite/python/treelite/../../build/libtreelite.so(TreeliteLoadXGBoostJSON+0x26) [0x7f0a4d6fc266]
E             [bt] (3) /home/phcho/miniconda3/lib/python3.8/lib-dynload/../../libffi.so.7(+0x69dd) [0x7f0a6e3f39dd]
E             [bt] (4) /home/phcho/miniconda3/lib/python3.8/lib-dynload/../../libffi.so.7(+0x6067) [0x7f0a6e3f3067]
E             [bt] (5) /home/phcho/miniconda3/lib/python3.8/lib-dynload/_ctypes.cpython-38-x86_64-linux-gnu.so(+0x10da8) [0x7f0a6e409da8]
E             [bt] (6) /home/phcho/miniconda3/lib/python3.8/lib-dynload/_ctypes.cpython-38-x86_64-linux-gnu.so(+0x1108c) [0x7f0a6e40a08c]
E             [bt] (7) python(_PyObject_MakeTpCall+0x22f) [0x5641add4485f]
E             [bt] (8) python(_PyEval_EvalFrameDefault+0x4596) [0x5641addcbf56]

python/treelite/core.py:50: TreeliteError
```

It appears that the ignore handler isn't properly registering for `feature_names` and `feature_types`.